components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/context/model/Flow.java

```diff
@@ -44,6 +44,18 @@ public class Flow {
                 EnumSet.of(InitiatingPersona.ADMIN, InitiatingPersona.APPLICATION));
         FLOW_DEFINITIONS.put(Name.GROUP_UPDATE,
                 EnumSet.of(InitiatingPersona.ADMIN, InitiatingPersona.APPLICATION));
+        FLOW_DEFINITIONS.put(Name.LOGOUT,
+                EnumSet.of(InitiatingPersona.APPLICATION, InitiatingPersona.USER));
+        FLOW_DEFINITIONS.put(Name.DELETE_USER,
+                EnumSet.of(InitiatingPersona.ADMIN, InitiatingPersona.APPLICATION));
+        FLOW_DEFINITIONS.put(Name.UPDATE_CREDENTIAL_PASSWORD,
+                EnumSet.of(InitiatingPersona.ADMIN, InitiatingPersona.APPLICATION, InitiatingPersona.USER));
+        FLOW_DEFINITIONS.put(Name.ACCOUNT_DISABLE,
+                EnumSet.of(InitiatingPersona.ADMIN, InitiatingPersona.USER, InitiatingPersona.APPLICATION));
+        FLOW_DEFINITIONS.put(Name.ACCOUNT_LOCK,
```

Comment:
The account lock action is not initiated by user, rather it is derived from user action. I feel like we have to add another persona here. Lets verify this.

Reply:
account lock can be system -> looks like we haven't onboarded that persona still. It can happen due to a policy
So let's onboard a persona as SYSTEM
account disable is admin or application ideally. let's remove user from there